src/google/protobuf/compiler/objectivec/method_dump.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+# Updates objectivec_nsobject_methods.h by generating a list of all of the properties
+# and methods on NSObject that Protobufs should not overload from iOS and macOS combined.
+#
+# The rules:
+#   - No property should ever be overloaded.
+#   - Do not overload any methods that have 0 args such as "autorelease".
+#   - Do not overload any methods that start with "set" and have 1 arg such as
+#     "setValuesForKeysWithDictionary:". Note that these will end up in the list as just
+#     the "proto field" name, so "setValuesForKeysWithDictionary:" will become
+#     "valuesForKeysWithDictionary".
+
+set -eu
+
+objc_code=$(cat <<'END_CODE'
+#import <Foundation/Foundation.h>
+#import <objc/runtime.h>
+
+int main(int argc, const char * argv[]) {
+  @autoreleasepool {
+    Class cls = [NSObject class];
+
+    // List out the protocols on NSObject just so we are aware if they change.
+    unsigned int protocolCount;
+    __unsafe_unretained Protocol **protocols =
+        class_copyProtocolList(cls, &protocolCount);
+    for (unsigned int i = 0; i < protocolCount; i++) {
+      printf("// Protocol: %s\n", protocol_getName(protocols[i]));
+    }
+    free(protocols);
+
+    // Grab all the properties.
+    unsigned int propCount;
+    objc_property_t *props = class_copyPropertyList(cls, &propCount);
+    NSMutableSet *reservedNames = [[NSMutableSet alloc] init];
+    for (unsigned int i = 0; i < propCount; ++i) {
+      NSString *propertyName = [NSString stringWithUTF8String:property_getName(props[i])];
+      [reservedNames addObject:propertyName];
+    }
+    free(props);
+
+    // Note that methods have 2 defaults args (_cmd and SEL) so a method "0 arg method"
+    // actually has 2.
+    unsigned int methodCount;
+    Method *methods = class_copyMethodList(cls, &methodCount);
```

Comment:
does this include methods from categories that are always added? For example, `accessibilityLabel`

Reply:
It isn't in the header in the PR, but I don't think it is because the categories aren't captures, I think it is because the test binaries are only linking Foundation.  We likely need to link in Cocoa/UIKit on each platform to ensure a more complete list of category additions are picked up.

Reply:
Done